### PR TITLE
Make icicle-bench the official icicle executable

### PR DESCRIPTION
**Pull request by jystic**
- Adds all the normal Ambiata executable features like `--version`, `--dependencies`
- Adds a standalone 'icicle compile' command
- The previous `icicle-bench` becomes 'icicle query'

Note that `icicle compile` requires some but not all of the options that `icicle query` requires:
```
$ icicle compile --help
Usage: icicle compile
    --dictionary-toml DICTIONARY_TOML
    --output-code DICTIONARY_C
    (--input-sparse-psv | --input-dense-psv | --input-zebra)
    (--output-sparse-psv | --output-dense-psv)
    (--snapshot SNAPSHOT_DATE | --chord)
  Compile a dictionary to its C intermediate form.

$ icicle query --help
Usage: icicle query
    (--dictionary-toml DICTIONARY_TOML | --dictionary-code DICTIONARY_C)
    (--input-sparse-psv INPUT_PSV | --input-dense-psv INPUT_PSV | --input-zebra INPUT_ZEBRA)
    (--output-sparse-psv OUTPUT_PSV | --output-dense-psv OUTPUT_PSV)
    [--output-code DICTIONARY_C]
    (--snapshot SNAPSHOT_DATE | --chord CHORD_DESCRIPTOR)
    [--facts-limit ARG]
    [--drop DROP_FILE]
    [--drop-to-output]
  Run an icicle query over some data.
```

@nhibberd we'll want to change payload `s/icicle-bench/icicle/`

! @amosr @tranma @nhibberd

### CODE REVIEW

**Comment by amosr** [comment]
🐒  👨

**Comment by tranma** [comment]
thank you optparse-man :bat:

File: icicle-compiler/src/Icicle/Command.hs [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+compileDictionary ::
+     FilePath
+  -> InputFormat
+  -> OutputFormat
+  -> Scope a
+  -> EitherT IcicleError IO Text
+compileDictionary dictionaryPath iformat oformat scope = do
+  -- FIXME We really need to include InputFormat/OutputFormat/Scope in the compiled
+  -- FIXME code so that we don't accidentally run with the wrong options.
```
Comment (tranma) [inline icicle-compiler/src/Icicle/Command.hs]: how do you mean? just define a few flags in the output C code?

Comment (jystic) [inline icicle-compiler/src/Icicle/Command.hs]: yeah, or maybe better, not actually output C code, but a tarball with the C code and a `.json` file that describes the options a file was compiled with. Then we can check that we're not going to explode when we try to run the snapshot.

**Pull request merged**